app/cards/Card.jsx

```diff
@@ -31,6 +31,17 @@ export default class Card extends PureComponent {
   }
 
   componentDidMount() {
+    this.getCard(this.props.params.id);
+  }
+
+  componentWillReceiveProps(nextProps) {
```

Comment:
This is what fixes the bug.

In the current implementation, when a card component is already rendered, clicking a card in the search bar has no effect because it only checked on mount for what the route params were.